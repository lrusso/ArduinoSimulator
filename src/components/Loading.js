import React from "react"
import Spinner from "./Spinner"

const Loading = () => {
  return (
    <div style={styles.container}>
      <div style={styles.icon}>
        <Spinner />
      </div>
    </div>
  )
}

const styles = {
  container: {
    position: "fixed",
    minWidth: "320px",
    left: "0",
    top: "0",
    right: "0",
    bottom: "0",
    backgroundColor: "rgba(0,0,0,0.5)",
    zIndex: "8888",
  },
  icon: {
    position: "absolute",
    left: "0",
    top: "50%",
    right: "0",
    marginLeft: "auto",
    marginRight: "auto",
    width: "64px",
    height: "64px",
    transform: "translateY(-50%)",
  },
}

export default Loading
