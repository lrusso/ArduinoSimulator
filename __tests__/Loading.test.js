import React from "react"
import renderer from "react-test-renderer"
import Loading from "../src/components/Loading"

describe("Loading", () => {
  test("Rendering 1", () => {
    const tree = renderer.create(<Loading />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
