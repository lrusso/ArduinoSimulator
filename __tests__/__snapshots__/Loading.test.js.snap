// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Rendering 1 1`] = `
<div
  style={
    {
      "backgroundColor": "rgba(0,0,0,0.5)",
      "bottom": "0",
      "left": "0",
      "minWidth": "320px",
      "position": "fixed",
      "right": "0",
      "top": "0",
      "zIndex": "8888",
    }
  }
>
  <div
    style={
      {
        "height": "64px",
        "left": "0",
        "marginLeft": "auto",
        "marginRight": "auto",
        "position": "absolute",
        "right": "0",
        "top": "50%",
        "transform": "translateY(-50%)",
        "width": "64px",
      }
    }
  >
    <div
      className="spinner_white_animation"
    >
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
      <div />
    </div>
  </div>
</div>
`;
